### PR TITLE
667 Use PHPUnit 7 for testing

### PR DESCRIPTION
**Pull request by MPParsley**
PHPUnit 7 can be used for testing on PHP 7.1+

https://www.drupal.org/node/3078763

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -71,6 +71,9 @@ before_script:
   # Install Composer dependencies for core. Skip this for the coding standards test.
   - test ${TEST_SUITE} == "PHP_CodeSniffer" || composer install --working-dir=$DRUPAL_DIR
 
+  # Update PHP
+  - test ${TEST_SUITE} == "PHP_CodeSniffer" || travis_retry composer update phpunit/phpunit symfony/phpunit-bridge phpspec/prophecy symfony/yaml --with-dependencies
```
Comment (MPParsley) [inline .travis.yml]:
```suggestion
  - test ${TEST_SUITE} == "PHP_CodeSniffer" || travis_retry composer update phpunit/phpunit symfony/phpunit-bridge phpspec/prophecy symfony/yaml --with-dependencies --working-dir=$DRUPAL_DIR
```

**Comment by MPParsley** [comment]
```

Job | PHP | ENV | OS | State | Notes
-- | -- | -- | -- | -- | --
2568.1 | 7.2 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.2 | 7.2 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.3 | 7.3 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.4 | 7.3 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.5 | 7.3 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.6 | 7.3 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.7 | 7.3 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed |  
2568.8 | 7.4 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed | This job is allowed to fail.
2568.9 | 7.4 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed | This job is allowed to fail.
2568.10 | 7.4 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed | This job is allowed to fail.
2568.11 | 7.4 | COMPOSER_MEMORY_LIMIT=2G | Linux | passed | This job is allowed to fail.


```

**Comment by pfrenssen** [comment]
Are there benefits to using PHPUnit 7 on Drupal 8? From what I have seen in Drupal 9 is that some functionality in PHPUnit 6 is deprecated and replaced with other classes / methods in PHPUnit 7, but AFAIK we are not using any of those.

This way of doing `composer require` after dependencies are already installed is quite inefficient. It adds ~ 1 minute of extra time to the Drupal 8 tests. The Drupal 9 builds are already using PHPUnit 7, but there we also get a 30 second time penalty because composer recalculates the dependency tree.

Maybe it is better to wait with this until we start using PHPUnit 7 specific functionality and this becomes necessary?

**Comment by MPParsley** [comment]
Indeed, it depends. I needed it because I had some tests that were using deprecated phpunit functions.

Let's close this for now.

**Comment by MPParsley** [comment]
@pfrenssen, the difference is that with this patch all tests are green while PHP 7.4 tests are now failing:

```
2531.8 | 7.4 | COMPOSER_MEMORY_LIMIT=2G | Linux | failed | This job is allowed to fail.
-- | -- | -- | -- | -- | --
2531.9 | 7.4 | COMPOSER_MEMORY_LIMIT=2G | Linux | failed | This job is allowed to fail.
```

The builds fail due to these unsilenced deprecation notices:
```

  10x: Function ReflectionType::__toString() is deprecated
    10x in GroupSubscribeFormatterTest::setUp from Drupal\Tests\og\Unit
 > Executing tests from /home/travis/build/Gizra/drupal/modules/og/og_ui/tests
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.
Testing /home/travis/build/Gizra/drupal/modules/og/og_ui/tests
..                                                                  2 / 2 (100%)
Time: 17.44 seconds, Memory: 6.00MB

```
See https://travis-ci.com/github/Gizra/og/jobs/362763095

**Comment by pfrenssen** [comment]
That's great news, if we can guarantee the correct working on PHP 7.4 then this is definitely worth the 1 minute time penalty. Reopening.

File: .travis.yml [hunk 2]
```diff
@@ -71,6 +71,9 @@ before_script:
   # Install Composer dependencies for core. Skip this for the coding standards test.
   - test ${TEST_SUITE} == "PHP_CodeSniffer" || composer install --working-dir=$DRUPAL_DIR
 
+  # Update PHP
```
Comment (MPParsley) [inline .travis.yml]:
```suggestion
  # Update PHPUnit.
```

**Review by pfrenssen — APPROVED** [review]

**Pull request merged**